src/gui/sdlmain.cpp

```diff
@@ -4567,8 +4567,9 @@ void restart_program(std::vector<std::string> & parameters) {
 
 	const WINBOOL inherit_handles = FALSE;
 
-	// DETACHED_PROCESS fixes https://github.com/dosbox-staging/dosbox-staging/issues/3346
-	const DWORD creation_flags = DETACHED_PROCESS;
+	// CREATE_NEW_CONSOLE fixes a bug where the parent process exiting kills the child process.
```

Comment:
Very economical with words, @weirddan455 😄 Can you add this at least to explain the "why" behind it?

```
This can manifest when you use the "restart" hotkey action to restart DOSBox.
```

Reply:
Thanks man 😄 Just thinking of future maintainers, that's all.